cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_utils.go

```diff
@@ -445,3 +465,21 @@ func (f *ConfigurableConfigFactoryImpl) ConfigureRESTConfig(config *rest.Config)
 func (f *ConfigurableConfigFactoryImpl) RESTConfig() (*rest.Config, error) {
 	return f.config, nil
 }
+
+func WaitForResource(ctx context.Context, ri dynamic.ResourceInterface, name string, interval, timeout time.Duration) error {
```

Comment:
I've extracted this fn to this utils pkg, but if it were to be useful for other plugins I can move it.

Using a dynamic resource interface so that we just have to build the GRV and get the resource from a dynamic client.

Reply:
Nice!

Reply:
Yes, I think it could be useful for other plugins (eg. for #4213)

Reply:
Great, I'm on in (in a separate PR, since I wanna move a couple of functions as well)